### PR TITLE
Add func. tests for Med service

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#9613](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9613)

* [ ] Enhancement
* [X] Bug
* [ ] Documentation

## Description
image

## Testing

* [ ] Unit Tests Updated
* [X] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes
NO

### Browsers Tested

* [X] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Testing/functional/e2e/cypress/integration/services/medication.js [hunk 1]
```diff
@@ -56,25 +104,25 @@ describe("Medication Service", () => {
 
     it("Verify MedicationStatement Authorized", () => {
         const HDID = "P6FFO433A5WPMVTGM7T4ZVWBKCSVNAYGTWTU3J2LWMGUMERKI72A";
-        cy.fixture("MedicationService/medicationStatement.json").then((MedicationStatementResponse) => {
```
Comment (tiagograf-AOT) [inline Testing/functional/e2e/cypress/integration/services/medication.js:59]: this fixture needs to be deleted if its not used anymore

File: Testing/functional/e2e/cypress/integration/services/medication.js [hunk 2]
```diff
@@ -16,6 +23,47 @@ describe("Medication Service", () => {
...
+            cy.request({
+                url: `${config.serviceEndpoints.Medication}v1/api/Medication/?${drugIdentifiersQueryString}`,
+                followRedirect: false,
+                headers: {
+                    accept: "application/json",
+                },
+            }).should((response) => {
+                expect(response.status).to.eq(200);
+                expect(response.body).to.not.be.null;
+                cy.log(`response.body: ${JSON.stringify(response.body)}`);
```
Comment (sslaws) [inline Testing/functional/e2e/cypress/integration/services/medication.js:40]: FYI:  You don't have to stringify instead you can simply log the object and inspect it in the console.

**Review by sslaws — DISMISSED** [review]

File: Testing/functional/e2e/cypress/integration/services/medication.js [hunk 3]
```diff
@@ -16,6 +23,47 @@ describe("Medication Service", () => {
...
+                verifyDrug(response.body.resourcePayload);
+            });
+        });
```
Comment (tiagograf-AOT) [inline Testing/functional/e2e/cypress/integration/services/medication.js:65]: we should also test a drug from the federal source (DIN) and not only the provincial (PIN)

**Review by tiagograf-AOT — CHANGES_REQUESTED** [review]
Missing tests, unused file.

File: Testing/functional/e2e/cypress/integration/services/medication.js (hunk 2 above)
Comment (leobcgov) [reply Testing/functional/e2e/cypress/integration/services/medication.js:40]: > FYI: You don't have to stringify instead you can simply log the object and inspect it in the console.
@sslaws good to know, as I saw only [Object Objec] in the console so I though I needed the stringify.

File: Testing/functional/e2e/cypress/integration/services/medication.js (hunk 3 above)
Comment (leobcgov) [reply Testing/functional/e2e/cypress/integration/services/medication.js:65]: > we should also test a drug from the federal source (DIN) and not only the provincial (PIN)
@tiagograf-AOT good point, I didn't know we had 2 types :-)

File: Testing/functional/e2e/cypress/integration/services/medication.js (hunk 1 above)
Comment (leobcgov) [reply Testing/functional/e2e/cypress/integration/services/medication.js:59]: removed the json

File: Testing/functional/e2e/cypress/integration/services/medication.js (hunk 2 above)
Comment (leobcgov) [reply Testing/functional/e2e/cypress/integration/services/medication.js]: removed stringify

File: Testing/functional/e2e/cypress/integration/services/medication.js (hunk 3 above)
Comment (leobcgov) [reply Testing/functional/e2e/cypress/integration/services/medication.js:93]: added test for FED

**Review by tiagograf-AOT — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**